docs/src/developer/design/overview.rst

```diff
@@ -223,41 +258,26 @@ Summary
    written entirely in CMake.
 
 :ref:`cmake_based_workflows`
-   Consumers interact with ``CMakeLists.txt`` written with CMaize no differently
-   than they would with ``CMakeLists.txt`` written with CMake alone. Therefore,
-   CMaize-based build systems seamlessly integrate into existing workflows.
-
-:ref:`object_oriented`
-   CMaize has adopted the `CMakePP Language`_ under the hood.
+   CMaize is distributed as a CMake module and is designed to be included in
+   a ``CMakeLists.txt`` file. In turn, the person or tool building the project
+   still interacts with the project using the traditional CMake-based workflow.
+
+:ref:`minimize_redundancy`
+   The user API is charged with collecting the package's information via a
+   series of functional-style calls and mapping it to the traditional CMake
+   calls. So although all of CMake's usual verboseness and redundancy is still
+   present, it is encapsulated under CMaize and the developer's build system
+   obeys :term:`DRY`.
+
+:ref:`target_support`
+   CMaize's Target component is responsible for being able to generically
+   handle any build targets the user may want to define. These include external
+   packages as well as build targets needed by the current project.
 
 :ref:`package_manager_support`
-   All APIs dealing with build targets allow the user to provide a package
-   manager.
-
-:ref:`project_meta_data`
-   This consideration primarily impacts CMaize in that build system developers
-   will have to do it in CMake directly.
-
-:ref:`obtain_cmaize`
-   Like :ref:`project_meta_data`, this step primarily impacts CMaize in that
-   it can not be abstracted away and must be present in the boilerplate.
-
-:ref:`declare_build_options`
-   For version 1.0.0 of CMaize we advocate for using CMake's
-   `option <https://tinyurl.com/529f5zn7>`_ command. In later versions of
-   CMaize we may decide to capture these options in the ``PackageSpecification``
-
-:ref:`find_dependencies`
-   This responsibility will ultimately be punted to the ``PackageManager``,
-   though we must provide the user a functional API to pass the info to the
-   ``PackageManager``. We propose the ``cmaize_find_or_build_dependency``
-   commands.
-
-:ref:`define_project_components`
-   ``cmaize_add_xxx`` commands have been proposed for these purposes.
+   Interacting with CMake's internal package manager, as well as external
+   package managers, is covered by the PackageManager component.
 
-:ref:`test_project_components`
-   ``cmaize_add_tests`` command has been proposed for this.
-
-:ref:`install_the_project`
-    ``cmaize_add_package`` command is responsible for this.
+:ref:`object_oriented`
+   The guts of CMaize is object-oriented. Internally, CMaize has adopted the
```

Comment:
```suggestion
   The guts of CMaize are object-oriented. Internally, CMaize has adopted the
```
Consider using "internals", "backend", or another synonym instead of "guts" here.